inc/packages/admin/namespace.php

```diff
@@ -0,0 +1,257 @@
+<?php
+/**
+ * Packages admin bootstrap.
+ *
+ * @package FAIR
+ */
+
+namespace FAIR\Packages\Admin;
+
+use FAIR;
+use FAIR\Packages;
+use FAIR\Packages\MetadataDocument;
+use FAIR\Packages\ReleaseDocument;
+use FAIR\Updater;
+
+use WP_Upgrader_Skin;
+
+const TAB_DIRECT = 'fair_direct';
+const ACTION_INSTALL = 'fair-install-plugin';
+const ACTION_INSTALL_NONCE = 'fair-install-plugin';
+
+/**
+ * Bootstrap.
+ */
+function bootstrap() {
+	if ( ! is_admin() ) {
+		return;
+	}
+
+	add_filter( 'install_plugins_tabs', __NAMESPACE__ . '\\add_direct_tab' );
+	add_action( 'install_plugins_' . TAB_DIRECT, __NAMESPACE__ . '\\render_tab_direct' );
+	add_action( 'load-plugin-install.php', __NAMESPACE__ . '\\load_plugin_install' );
+	add_action( 'install_plugins_pre_plugin-information', __NAMESPACE__ . '\\maybe_hijack_plugin_info', 0 );
+	add_action( 'update-custom_' . ACTION_INSTALL, __NAMESPACE__ . '\\handle_direct_install' );
+}
+
+/**
+ * Filters the tabs shown on the Add Plugins screen.
+ *
+ * @param string[] $tabs Map of tab ID to map name.
+ * @return string[] Updated tabs.
+ */
+function add_direct_tab( $tabs ) {
+	$tabs[ TAB_DIRECT ] = __( 'Direct Install', 'fair' );
+	return $tabs;
+}
+
+/**
+ * Enqueue assets.
+ *
+ * @return void
+ */
+function load_plugin_install() {
+	enqueue_assets();
+}
+
+/**
+ * Enqueue assets.
+ *
+ * @return void
+ */
+function enqueue_assets() {
+	wp_enqueue_style(
+		'fair-admin',
+		esc_url( plugin_dir_url( FAIR\PLUGIN_FILE ) . 'assets/css/packages.css' ),
+		[],
+		FAIR\VERSION
+	);
+}
+
+/**
+ * Render direct installer tab.
+ *
+ * @return void
+ */
+function render_tab_direct() {
+	?>
+	<div class="fair-direct-install">
+		<p class="fair-direct-install__help">
+			<?= __( "If you have a plugin's ID, enter it here to view the details<br />and install it.", 'fair' ); ?>
+		</p>
+		<form
+			action=""
+			class="fair-direct-install__form"
+			method="post"
+		>
+			<input
+				type="hidden"
+				name="tab"
+				value="<?= esc_attr( TAB_DIRECT ); ?>"
+			/>
+			<?php wp_nonce_field( TAB_DIRECT ); ?>
+			<label
+				class="screen-reader-text"
+				for="plugin_id"
+			>Plugin ID</label>
+			<input
+				type="text"
+				id="plugin_id"
+				name="plugin_id"
+				pattern="did:(web|plc):.+"
+				placeholder="did:..."
+			/>
+			<?php submit_button( _x( 'View Details', 'plugin', 'fair' ), '', '', false ); ?>
+		</form>
+		<p class="fair-direct-install__note">
```

Comment:
This text is important to knowing how to fill out the plugin ID field; let's give it an explicit association with the input field via `aria-describedby`.

Reply:
Done in [83e092f](https://github.com/fairpm/fair-plugin/pull/71/commits/83e092ff8e176ae585764153fcf4f27aa7007959).